src/servers/grpc_server.cc

```diff
@@ -150,9 +155,15 @@ AsyncResources::ResponseAlloc(
   // that the number and order of raw output entries equals the output
   // meta-data.
   std::string* raw_output = response->add_raw_output();
-  if (byte_size > 0) {
-    raw_output->resize(byte_size);
-    *buffer = static_cast<void*>(&((*raw_output)[0]));
+  auto pr = output_shm_map.find(tensor_name);
+  if (pr != output_shm_map.end()) {
+    *buffer =
```

Comment:
You should check to make sure that the requested byte size matches what is provided by the shared memory.

Reply:
Done